v2/js/components/listview.js

```diff
@@ -0,0 +1,282 @@
+/* @ListView
+ * @desc: ListView is a class that is used to create a list view screen.
+ */
+
+const ListView = {
+	name: 'ListView',
+	template: ` <div class="listview" v-for="activity in sortObjectArray(activities)" :key="activity.id">
+						<div class="listview_left" >
+							<icon
+								:ref="'star' + activity.id"
+								:id="'star' + activity.id"
+								svgfile="./icons/star.svg"
+								:color="getStarColor(activity)"
+								size="22"
+								:x=0
+								:y=0
+								@click="toggleFavorite(activity)"
+								style="padding: 10px;"
+							></icon>
+							<div style="width:44px">
+								<icon 
+									:id=activity.id
+									:svgfile="activity.directory + '/' + activity.icon"
+									size="40"
+									isNative="true"
+									v-on:mouseover="showPopupFunction($event)"
+									v-on:mouseleave="removePopupFunction($event)"
+									@click="launchActivity(activity)"
+									style="padding: 2px;"
+								></icon>
+								</div>
+							
+							<div class="activity-name">{{ activity.name }}</div>
+						</div>
+						<div class="activity-version">Version {{ activity.version }}</div>
+						<icon 
+							:id="'help' + activity.id"
+							svgfile="./icons/help-rev.svg"
+							color="256"
+							size="44"
+							isNative="true"
+						></icon>
+					</div>
+					<popup 
+						ref="popup" 
+						v-bind:item="popup"
+						v-on:mouseleave="removePopupFunction($event)"
+						v-on:itemis-clicked="itemisClicked($event)"
+					></popup>
+				`,
+
+	components: {
+		'icon': Icon,
+		'popup': Popup,
+	},
+
+	emits: ['activities'],
+
+	data() {
+		return {
+			token: null,
+			favactivities: [],
+			activities: [],
+			popupData: null,
+			popup: null, //singular popup data
+			buddycolor: null,
+		}
+	},
+
+	props: ['filteredactivities'],
+
+	mounted() {
+		this.token = JSON.parse(localStorage.getItem("sugar_settings")).token;
+		this.getActivities();
+	},
+
+	watch: {
+		filteredactivities: function (value) {
```

Comment:
Fix typo

Reply:
Typo? It's not a typo we are watching over filteredactivities prop here. And updating activities array according to it.